src/realm/sync/noinst/client_history_impl.cpp

```diff
@@ -399,33 +399,36 @@ void ClientHistory::integrate_server_changesets(const SyncProgress& progress,
                 changesets[i].last_integrated_remote_version = m_sync_history_base_version;
         }
 
-        Transformer& transformer = get_transformer(); // Throws
-        Transformer::Reporter* reporter = nullptr;
-        transformer.transform_remote_changesets(*this, transact->get_sync_file_id(), local_version, changesets.data(),
-                                                changesets.size(), reporter, &logger); // Throws
-
-        for (std::size_t i = 0; i < num_changesets; ++i) {
-            ChangesetEncoder::Buffer transformed_changeset;
-            encode_changeset(changesets[i], transformed_changeset);
+        if (m_replication.apply_server_changes()) {
```

Comment:
This conditional is the only actual change here; everything else is just reindenting and the sync client changes are all just passing down the bool to be able to do this check.